server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -172,18 +173,22 @@ private void confirmDeletes(GarbageCollectionEnvironment gce,
 
     }
 
+    Set<String> tableIdsBefore = gce.getTableIDs();
+    Set<String> tableIdsSeen = new HashSet<>();
     Iterator<Entry<Key,Value>> iter = gce.getReferenceIterator();
     while (iter.hasNext()) {
       Entry<Key,Value> entry = iter.next();
       Key key = entry.getKey();
       Text cft = key.getColumnFamily();
+      tableIdsSeen.add(new String(KeyExtent.tableOfMetadataRow(key.getRow())));
 
       if (cft.equals(DataFileColumnFamily.NAME) || cft.equals(ScanFileColumnFamily.NAME)) {
         String cq = key.getColumnQualifier().toString();
 
         String reference = cq;
         if (cq.startsWith("/")) {
           String tableID = new String(KeyExtent.tableOfMetadataRow(key.getRow()));
+          tableIdsSeen.add(tableID);
```

Comment:
Seems like the code is deriving the table id from the row multiple times and adding it (or is there some little diff I am missing?).  May be able to do the following.  

```suggestion
      String tableID = new String(KeyExtent.tableOfMetadataRow(key.getRow()));
      tableIdsSeen.add(tableID);

      if (cft.equals(DataFileColumnFamily.NAME) || cft.equals(ScanFileColumnFamily.NAME)) {
        String cq = key.getColumnQualifier().toString();

        String reference = cq;
        if (cq.startsWith("/")) {
```

Reply:
This is a good change, I'll implement